pkg/declarativeconfig/uuid_test.go

```diff
@@ -0,0 +1,37 @@
+package declarativeconfig
+
+import (
+	"testing"
+
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeterministicUUIDs(t *testing.T) {
+	declarativeUUIDFuncs := []func(string) uuid.UUID{
+		NewDeclarativeAuthProviderUUID,
+		NewDeclarativeGroupUUID,
+		NewDeclarativePermissionSetUUID,
+		NewDeclarativeAccessScopeUUID,
+	}
+	dummyName := "dummy-resource"
+
+	// 1. Test that using the same name will lead to the same UUID being created.
+	for _, f := range declarativeUUIDFuncs {
+		firstID := f(dummyName)
+		secondID := f(dummyName)
+		assert.Equal(t, firstID, secondID)
+	}
+
+	// 2. Test that using the same name won't lead to clashes within the different namespaces.
```

Comment:
I suggest adding the third test: testing that different names in the same namespace result in different UUIDs. 
Otherwise, currently this test can be passed by `func constant(a string) { return "constantValue" }`